api/users/serializers.py

```diff
@@ -34,6 +35,27 @@ def to_representation(self, value):
         return relationship_links
 
 
+class SocialField(ser.DictField):
+    def __init__(self, min_version, **kwargs):
+        super(SocialField, self).__init__(min_version, **kwargs)
+        self.min_version = min_version
+        self.help_text = 'This field will change data formats after version {}'.format(self.min_version)
+
+    def to_representation(self, value):
+        old_social_string_fields = ['twitter', 'github', 'linkedIn']
+        request = self.context.get('request')
+        show_old_format = request and is_deprecated(request.version, self.min_version)
+        if show_old_format:
+            new_value = {}
+            for key, socialvalue in value.iteritems():
```

Comment:
Drive-by: It would be more straight-forward to loop through `old_social_string_field` rather than the entire `social` dictionary.

Reply:
Just make sure to make a copy of the social dictionary for the return value (rather than mutating it in-place.